docs/features/populate.rst

```diff
@@ -0,0 +1,296 @@
+        ]
+    }
+
+
+    @populate_with(data, context=True, verbose=1)
+    def test_org_1(context):
+        """a test with populated data"""
+        assert context.registry['organization_1'].name == "MyOrganization1"
+
+
+The YAML data file
+------------------
+
+In the YAML data file it is possible to specify 3 sections, ``config``, ``vars`` and ``actions``.
+
+
+config
+++++++
+
+The config may be used to define special behavior of populator and its keys are:
+
+- populator
+
+  The name of the populator defined in ``populators``
+- populators
+
+  The specification of populator modules to be loaded
+- verbose
+
+  The verbosity of logging 0, 1 or 2, it can be overwritten with -vvv in commands.
+
+example:
+
+.. code-block:: console
+
+    config:
+      verbose: 3
+      populator: api
+      populators:
+        api:
+          module: robottelo.populate.api.APIPopulator
+        cli:
+          module: robottelo.populate.cli.CLIPopulator
+
+vars
+++++
+
+Variables to be available in the rendering context of the YAML data
+every var defined here is available to be referenced using ``Jinja`` syntax in
+any action.
+
+.. code-block:: console
+
+      vars:
+        admin_username: admin
+        admin_password: changeme
+        org_name_list:
+          - company7
+          - company8
+        prefix: aaaa
+        suffix: bbbb
+        my_name: me
+
+actions
++++++++
+
+The actions is the most important section of the YAML, it is a list of actions
+being each action a dictionary containing special keys depending on the action type.
+
+The action type is defined in ``action`` key and available actions are:
+
+Actions are executed in the defined order and order is very important because
+each action can ``register`` its result to the internal registry to be referenced
+later in any other action.
+
+
+**CRUD ACTIONS**
+
+Crud actions takes a ``model`` argument, any from ``nailgun.entities`` is valid.
+
+
+- create (the default)
+
+  Creates a new entity if not exists, else gets existing.
+- update
+
+  Updates entity with provided ``data`` by ``id`` or unique search
+- delete
+
+  deleted entity with ``id`` or unique search
+
+**SPECIAL ACTIONS**
+
+- echo
+
+  Logs and print output to the console
+- register
+
+  Register a variable in the internal registry
+- unregister
+
+  removes a variable from register
+- assertion
+
+  perform assertion operations, if any fails returns exit code 1
+
+Dynamic Data
+++++++++++++
+
+There are some ways to fetch dynamic data in action definitions, it depends
+on the action type.
+
+For any key you can use ``Jinja`` to provide a dynamic value as in::
+
+  value: "{{ get_something }}"
+  value: "{{ fauxfactory.gen_string('alpha')}}"
+  value: user_{{ item }}
+
+For some actions you can provide a ``data`` key, that data is used to create
+new entities and also to perform searches or build the action function.
+
+Every ``data`` key accepts 4 special reference directives in its sub-keys.
+
+- from_registry
```

Comment:
That has a reason. 

`from_registry` accepts pure string and you can write `var_name[0].property.object.function(a=1, b=2)` etc.. you are not limited, you can write Python code in there and will get the result.

`from_object` expects a module name, can be anything `polarion.foo.bar.zaz` and this name is imported dynamically, but if you need specific key or attribute or need to call a function you can pass `key, attr` parameters to get that.